edx_api/enrollments/__init__.py

```diff
@@ -22,6 +25,88 @@ def __init__(self, requester, base_url):
         """
         self.requester = requester
         self.base_url = base_url
+        self._course_details = {}
+
+    def _get_course_details(self, course_id):
+        """
+        Retrieves course details.
+
+        Args:
+            course_id (str): Course id.
+        """
+        if course_id in self._course_details:
+            return self._course_details[course_id]
+
+        req_url = urljoin(self.base_url, self.course_details_url.format(course_key=course_id))
```

Comment:
See comment below for another reason and how we could remove this, but I think all of these calls are unnecessary, and could wreck network performance.
Imagine a case where there are 100 or more different courses returned for one call, now in addition to the single request to the enrollments api, we're also creating 100 other requests to this courses api.